libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -437,16 +437,22 @@ void DeclarationTypeChecker::endVisit(VariableDeclaration const& _variable)
 		type = TypeProvider::withLocation(ref, typeLoc, isPointer);
 	}
 
-	if (
-		_variable.isConstant() &&
-		!dynamic_cast<UserDefinedValueType const*>(type) &&
-		type->containsNestedMapping()
-	)
-		m_errorReporter.fatalDeclarationError(
-			3530_error,
-			_variable.location(),
-			"The type contains a (nested) mapping and therefore cannot be a constant."
+	if (_variable.isConstant())
+	{
+		if (!type->isValueType())
+		{
+			bool allowed = false;
+			if (auto arrayType = dynamic_cast<ArrayType const*>(type))
+				allowed = arrayType->isByteArray();
+			if (!allowed)
+				m_errorReporter.fatalTypeError(9259_error, _variable.location(), "Constants of non-value type not yet implemented.");
+		}
+
+		solAssert(
+			!dynamic_cast<UserDefinedValueType const*>(type) &&
+			type->containsNestedMapping()
```

Comment:
This doesn't look right - you meant this negated, right?

Reply:
Oh. Haha, yeah. Famous last minute changes. Before it would complain that the error code is in no test, so I figured make it an assert if it can't ever happen anyway.

Reply:
Or rather ``solAssert(dynamic_cast<UserDefinedValueType const*>(type) || !type->containsNestedMapping());``?
Not sure we need it at all, though, since we assert that it's a value type above anyways.